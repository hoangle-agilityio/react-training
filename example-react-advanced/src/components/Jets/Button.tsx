import { memo } from 'react';

interface ButtonProps {
  buttonName: string;
  className: string;
  onClick?: () => void;
}

const Button = ({ buttonName, className, onClick }: ButtonProps): JSX.Element => {
  return (
    <button data-testid="button" className={className} onClick={onClick}>
      <span className="button-title">{buttonName}</span>
    </button>
  );
};

export default memo(Button);
