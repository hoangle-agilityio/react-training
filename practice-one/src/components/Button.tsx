interface ButtonProps {
  typeButton: "button" | "submit" | "reset";
  size: "sm" | "md"| "lg";
  buttonColor: "light" | "info" | "danger";
  customize?: string;
  label: string;
}

export default function Button({ typeButton, size, buttonColor, customize, label }: ButtonProps) {
  const separateClass = customize ? customize : "";

  return (
    <button
      type={typeButton}
      className={`${separateClass} btn btn-${buttonColor} btn--${size}`}
    >
      {label}
    </button>
  );
}
