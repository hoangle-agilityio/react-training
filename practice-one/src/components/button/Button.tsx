import "./button.scss";

interface ButtonProps {
  typeButton: "button" | "submit" | "reset";
  size?: "sm" | "md"| "lg";
  buttonColor: "light" | "info" | "danger";
  customize?: string;
  label: string;
}

export default function Button({ typeButton, size, buttonColor, customize, label }: ButtonProps) {
  const separateClass: string = customize ? customize : "";
  const buttonSize: string = size ? `btn--${size}` : "";

  return (
    <button
      type={typeButton}
      className={`${separateClass} btn btn-${buttonColor} ${buttonSize}`}
    >
      {label}
    </button>
  );
}
