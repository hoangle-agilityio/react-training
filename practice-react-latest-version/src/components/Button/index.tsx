// Libs
import * as React from "react";
import { memo } from "react";
import { Button as ReactButton, ButtonProps } from "@chakra-ui/react";

type Props = {
  label: React.ReactNode;
  disabled?: boolean;
  type?: "primary" | "secondary" | "text";
  styles?: ButtonProps;
  leftIcon?: React.ReactElement<React.ReactNode>;
  rightIcon?: React.ReactElement<React.ReactNode>;
  onClick?: () => void;
};

const Button = ({
  label,
  disabled = false,
  type = "primary",
  styles,
  leftIcon,
  rightIcon,
  onClick,
}: Props) => {
  let props = null;
  switch (type) {
    case "primary":
      props = {
        bgColor: "blue.300",
        color: "white.100",
      };
      break;
    case "secondary":
      props = {
        bgColor: "transparent",
        color: "red.200",
      };
      break;
    default:
      props = {
        bgColor: "transparent",
      };
  }

  return (
    <ReactButton
      fontSize="base"
      fontFamily="medium"
      fontWeight="500"
      _hover={{
        opacity: "70%",
        bgColor: `${type === "secondary" ? "gray.200" : "transparent"}`,
      }}
      isDisabled={disabled}
      leftIcon={leftIcon}
      rightIcon={rightIcon}
      onClick={onClick}
      {...props}
      {...styles}
    >
      {label}
    </ReactButton>
  );
};

export default memo(Button);
