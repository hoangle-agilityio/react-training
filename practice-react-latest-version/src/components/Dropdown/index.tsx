// Libs
import { Menu, MenuButton, MenuList, MenuListProps } from "@chakra-ui/react";
import React, { memo } from "react";

type Props = {
  children: React.ReactNode;
  renderMenu: () => React.ReactNode;
  stylesOption?: MenuListProps;
};

const Dropdown = ({ children, renderMenu, stylesOption }: Props) => {
  return (
    <Menu>
      <MenuButton aria-label="toggle-button" fontSize="base">
        {renderMenu()}
      </MenuButton>
      <MenuList minWidth="100px" {...stylesOption}>
        {children}
      </MenuList>
    </Menu>
  );
};

export default memo(Dropdown);
