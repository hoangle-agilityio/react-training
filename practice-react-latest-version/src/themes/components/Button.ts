export const Button = {
  baseStyle: {
    fontFamily: "medium",
    fontWeight: "500",
    _hover: {
      opacity: "70%",
    },
  },
  variants: {
    primary: {
      bgColor: "background.primary",
      color: "brand.100",
      fontSize: "base",
    },
    secondary: {
      bgColor: "transparent",
      color: "text.reversal",
      fontSize: "base",
      _hover: {
        bgColor: "background.default",
      },
    },
  },
  defaultProps: {
    variant: "primary",
  },
};
