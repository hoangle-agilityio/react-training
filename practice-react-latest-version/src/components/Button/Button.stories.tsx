// Libs
import * as React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";

// Components
import Button from "components/Button";

export default {
  title: "Components/Button",
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  ...Primary.args,
  label: "+ Add customer",
  type: "primary",
};

export const Secondary = Template.bind({});
Secondary.args = {
  label: "Cancel",
  type: "secondary",
};

export const Text = Template.bind({});
Text.args = {
  label: "Edit",
  type: "text",
  styles: {
    color: "blue.200",
  },
};
