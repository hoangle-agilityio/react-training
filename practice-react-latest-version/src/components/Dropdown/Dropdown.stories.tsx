// Libs
import * as React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { Image } from "@chakra-ui/react";

// Components
import Dropdown from "components/Dropdown";
import Button from "components/Button";

export default {
  title: "Components/Dropdown",
  component: Dropdown,
} as ComponentMeta<typeof Dropdown>;

const Template: ComponentStory<typeof Dropdown> = (args) => {
  const renderMenu = () => <Button label="Open Dropdown" />;

  return (
    <Dropdown
      {...args}
      renderMenu={renderMenu}
      stylesOption={{ width: "120px" }}
    >
      <Button
        label="View"
        type="text"
        styles={{
          _hover: {
            bgColor: "gray.200",
          },
          width: "full",
          justifyContent: "space-between",
          borderRadius: "0",
          color: "blue.300",
        }}
        rightIcon={<Image src="/icons/info.svg" />}
      />
      <Button
        label="Edit"
        type="text"
        styles={{
          _hover: {
            bgColor: "gray.200",
          },
          width: "full",
          justifyContent: "space-between",
          borderRadius: "0",
          color: "blue.300",
        }}
        rightIcon={<Image src="/icons/pencil-filed.svg" />}
      />
      <Button
        label="Delete"
        type="text"
        styles={{
          _hover: {
            bgColor: "gray.200",
          },
          width: "full",
          justifyContent: "space-between",
          borderRadius: "0",
          color: "red.200",
        }}
        rightIcon={<Image src="/icons/trash-filled.svg" />}
      />
    </Dropdown>
  );
};

export const Default = Template.bind({});
Default.args = {
  ...Default.args,
};
