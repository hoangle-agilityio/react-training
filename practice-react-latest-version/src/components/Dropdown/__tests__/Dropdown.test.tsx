// Libs
import { render } from "@testing-library/react";
import * as React from "react";
import { Text } from "@chakra-ui/react";

// Components
import Dropdown from "..";
import Button from "components/Button";

const renderMenu = () => <Text>...</Text>;

const children: React.ReactNode = (
  <Button
    label="Edit"
    type="text"
    styles={{
      _hover: {
        bgColor: "gray.200",
      },
      width: "full",
      justifyContent: "space-between",
      borderRadius: "0",
      color: "blue.300",
    }}
  />
);

const props = {
  renderMenu,
  children,
};

describe("Dropdown Component", () => {
  test("Should match snapshoot for Dropdown", () => {
    const dom = render(<Dropdown {...props} />);

    expect(dom).toMatchSnapshot();
  });
});
