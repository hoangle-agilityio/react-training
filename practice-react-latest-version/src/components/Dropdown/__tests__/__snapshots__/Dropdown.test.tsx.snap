// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component Should match snapshoot for Dropdown 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        aria-controls="menu-list-:r1:"
        aria-expanded="false"
        aria-haspopup="menu"
        aria-label="toggle-button"
        class="chakra-menu__menu-button css-1rfdk2h"
        id="menu-button-:r1:"
      >
        <span
          class="css-xl71ch"
        >
          <p
            class="chakra-text css-0"
          >
            ...
          </p>
        </span>
      </button>
      <div
        class="css-0"
        style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
      >
        <div
          aria-orientation="vertical"
          class="chakra-menu__menu-list css-2vme5"
          id="menu-list-:r1:"
          role="menu"
          style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
          tabindex="-1"
        >
          <button
            class="chakra-button css-1kwrwyg"
            type="button"
          >
            Edit
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <button
      aria-controls="menu-list-:r1:"
      aria-expanded="false"
      aria-haspopup="menu"
      aria-label="toggle-button"
      class="chakra-menu__menu-button css-1rfdk2h"
      id="menu-button-:r1:"
    >
      <span
        class="css-xl71ch"
      >
        <p
          class="chakra-text css-0"
        >
          ...
        </p>
      </span>
    </button>
    <div
      class="css-0"
      style="visibility: hidden; position: absolute; min-width: max-content; inset: 0 auto auto 0;"
    >
      <div
        aria-orientation="vertical"
        class="chakra-menu__menu-list css-2vme5"
        id="menu-list-:r1:"
        role="menu"
        style="transform-origin: var(--popper-transform-origin); opacity: 0; visibility: hidden; transform: scale(0.8) translateZ(0);"
        tabindex="-1"
      >
        <button
          class="chakra-button css-1kwrwyg"
          type="button"
        >
          Edit
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
